distbench_engine.cc

```diff
@@ -805,18 +809,21 @@ void DistBenchEngine::ActionListState::CancelActivities() {
   }
 }
 
-std::map<std::string, ActivityLog>
-DistBenchEngine::ActionListState::GetAllActivitiesLog() {
+void DistBenchEngine::ActionListState::UpdateActivitiesLog(
+    std::map<std::string, ActivityLog>& activities_logs) {
   int size = action_list->proto.action_names_size();
-  std::map<std::string, ActivityLog> ret;
   for (int i = 0; i < size; ++i) {
     auto action_state = &state_table[i];
     if (action_state->action->proto.has_activity_config_name()) {
+      auto activity_name = action_state->action->proto.name();
```

Comment:
You are using the variable to hold a temporary, but then re-computing the exact same temporary below.

Reply:
Done